st2common/st2common/content/loader.py

```diff
@@ -204,6 +240,36 @@ def _get_folder(self, pack_dir, content_type):
             return None
         return path
 
+    def _get_runners_from_dir(self, base_dir):
+        result = {}
+        for runner_name in os.listdir(base_dir):
```

Comment:
On a related note, we have a similar "issue" with packs - we actually use pack directory name as pack name ID, even though there is a "name" attribute in the metadata file (the latter is only used for display in API).

It's slightly confusing, but yeah for consistency we should probably do the same for runners, but the problem is runners names right now contain dashes, but for consistency with packs, etc. we should only allow underscores in the directory name.

Anyway, we can probably come up with some compromise, similar as we do with pack name alias right now for backward compatibility.